File: 11-05-2024/Juggler Sequence.py
#User function Template for python3

class Solution:
    def jugglerSequence(self, n):
#Your Code Start
        sequence = [n] 
        
        while n != 1:
            if n % 2 == 0:
                n = int(n ** 0.5) 
            else:
                n = int((n ** 3) ** 0.5)  
            sequence.append(n)  
        
        return sequence
  #Your Code End





#{ 
 # Driver Code Starts
#Initial Template for Python 3

if __name__ == '__main__':
    t = int(input())
    for _ in range(t):
        n = int(input())

        ob = Solution()
        arr = ob.jugglerSequence(n)
        for i in (arr):
            print(i, end=" ")
        print()

# } Driver Code Ends

File: 11-05-2024/Juggler Sequence.cs
//{ Driver Code Starts
// Initial Template for C#

using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace DriverCode {

class GFG {
    static void Main(string[] args) {
        int testcases; // Taking testcase as input
        testcases = Convert.ToInt32(Console.ReadLine());
        while (testcases-- > 0) // Looping through all testcases
        {
            long n = Convert.ToInt64(Console.ReadLine()); // Input for size of array
            Solution obj = new Solution();
            List<long> res = obj.jugglerSequence(n);
            for (int i = 0; i < res.Count; i++) {
                Console.Write(res[i] + " ");
            }
            Console.Write("\n");
        }
    }
}

}

// } Driver Code Ends




// User function Template for C#
class Solution {
    // Complete this function
    public List<long> jugglerSequence(long n) {
      //Your Code Start
         List<long> sequence = new List<long>();
        sequence.Add(n); 
        
        while (n > 1) {
            if (n % 2 == 1) {
                n = (long)(Math.Sqrt(n) * n);
            } else {
                n = (long)Math.Sqrt(n); 
            }
            sequence.Add(n); 
        }
        
        return sequence;
        // Your Code End
    }
}
